src/ImageSharp.Drawing.Paths/FillPaths.cs

```diff
@@ -0,0 +1,87 @@
+﻿// <copyright file="FillPaths.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System;
+    using System.Numerics;
+    using Drawing;
+    using Drawing.Brushes;
+    using Drawing.Processors;
+
+    using SixLabors.Shapes;
+
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TColor}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Flood fills the image in the shape of the provided polygon with the specified brush..
+        /// </summary>
+        /// <typeparam name="TColor">The type of the color.</typeparam>
+        /// <param name="source">The source.</param>
+        /// <param name="brush">The brush.</param>
+        /// <param name="path">The shape.</param>
+        /// <param name="options">The graphics options.</param>
+        /// <returns>
+        /// The Image
+        /// </returns>
+        public static Image<TColor> Fill<TColor>(this Image<TColor> source, IBrush<TColor> brush, IPath path, GraphicsOptions options)
```

Comment:
`IPath` & `IShape` are interfaces from `SixLabors.Shapes` that's why we also have `Region` & `Drawable`(formally `Path`) which are what IPath & and Shape are wrapped into before filling/drawing happens